packages/rln/src/contract/rln_contract.spec.ts

```diff
@@ -1,82 +1,277 @@
-    const rlnContract = new RLNContract(rlnInstance, {
-      registryAddress: SEPOLIA_CONTRACT.address,
-      signer: voidSigner
+    const identity: IdentityCredential =
+      rlnInstance.zerokit.generateSeededIdentityCredential(mockSignature);
+
+    const insertMemberSpy = sinon.stub();
+    rlnInstance.zerokit.insertMember = insertMemberSpy;
+
+    const formatIdCommitment = (idCommitmentBigInt: bigint): string =>
+      "0x" + idCommitmentBigInt.toString(16).padStart(64, "0");
+
+    const membershipRegisteredEvent = mockRLNv2RegisteredEvent(
+      formatIdCommitment(identity.IDCommitmentBigInt)
+    );
+
+    const registerStub = sinon.stub().returns({
+      wait: () =>
+        Promise.resolve({
+          events: [
+            {
+              event: "MembershipRegistered",
+              args: {
+                idCommitment: formatIdCommitment(identity.IDCommitmentBigInt),
+                rateLimit: DEFAULT_RATE_LIMIT,
+                index: ethers.BigNumber.from(1)
+              }
+            }
+          ]
+        })
     });
 
-    let registerCalled = false;
-    rlnContract["storageIndex"] = 1;
-    rlnContract["_membersFilter"] = {
-      address: "",
-      topics: []
-    } as unknown as ethers.EventFilter;
-    rlnContract["registryContract"] = {
-      "register(uint16,uint256)": () => {
-        registerCalled = true;
-        return Promise.resolve({ wait: () => Promise.resolve(undefined) });
-      }
-    } as unknown as ethers.Contract;
-
-    const identity =
-      rlnInstance.zerokit.generateSeededIdentityCredential(mockSignature);
-    await rlnContract.registerWithIdentity(identity);
+    const mockedRegistryContract = {
+      register: registerStub,
+      queryFilter: () => [membershipRegisteredEvent],
+      provider: {
+        getLogs: () => [],
+        getBlockNumber: () => Promise.resolve(1000),
+        getNetwork: () => Promise.resolve({ chainId: 11155111 })
+      },
+      address: SEPOLIA_CONTRACT.address,
+      interface: {
+        getEvent: (eventName: string) => ({
+          name: eventName,
+          format: () => {}
+        })
+      },
+      filters: {
+        MembershipRegistered: () => ({}),
+        MembershipRemoved: () => ({})
+      },
+      on: () => ({}),
+      removeAllListeners: () => ({})
+    };
+
+    const provider = new ethers.providers.JsonRpcProvider();
+    const voidSigner = new ethers.VoidSigner(
+      SEPOLIA_CONTRACT.address,
+      provider
+    );
+    const rlnContract = await RLNContract.init(rlnInstance, {
+      signer: voidSigner,
+      address: SEPOLIA_CONTRACT.address,
+      rateLimit: DEFAULT_RATE_LIMIT,
+      contract: mockedRegistryContract as unknown as ethers.Contract
+    });
+
+    const decryptedCredentials =
+      await rlnContract.registerWithIdentity(identity);
+
+    expect(decryptedCredentials).to.not.be.undefined;
+    if (!decryptedCredentials)
+      throw new Error("Decrypted credentials should not be undefined");
```

Comment:
please, enclose `throw` into `{}` as on a it looks like `expect` is  right after `throw`, in particular on smaller screens